### PR TITLE
Update versions for 0.8.0 release

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Update versions for 0.8.0 release preparation.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,23 @@ All notable changes to this project will be documented in this file.
...
+
+### Removed
+
+### Fixed
+
+### Security
+
+### Dependencies
+
+## [0.8.0] - 2024-09-XX
```
Comment (NickGeneva) [inline CHANGELOG.md:25]: Update to target release day

File: CHANGELOG.md
Comment (NickGeneva) [inline CHANGELOG.md:1]: Also can you remove the empty sections with no details for 0.8.0

**Review by NickGeneva — APPROVED** [review]

File: CHANGELOG.md
Comment (ktangsali) [reply CHANGELOG.md:1]: Good catch. Done!

**Review by NickGeneva — APPROVED** [review]

**Pull request merged**